src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/AmendProbationBookingService.kt

```diff
@@ -0,0 +1,112 @@
+  private val prisonerValidator: PrisonerValidator,
+  private val additionalBookingDetailRepository: AdditionalBookingDetailRepository,
+) {
+  companion object {
+    private val log = LoggerFactory.getLogger(this::class.java)
+  }
+
+  @Transactional
+  fun amend(videoBookingId: Long, request: AmendVideoBookingRequest, amendedBy: User): Pair<VideoBooking, Prisoner> {
+    require(request.bookingType == BookingType.PROBATION) {
+      "PROBATION BOOKINGS: booking type is not probation"
+    }
+
+    val booking = videoBookingRepository.findById(videoBookingId)
+      .orElseThrow { EntityNotFoundException("Video booking with ID $videoBookingId not found.") }
+      .also { checkVideoBookingAccess(amendedBy, it) }
+      .also { checkCaseLoadAccess(amendedBy, it.prisonCode()) }
+      .also { require(BookingType.valueOf(it.bookingType.name) == request.bookingType) { "The booking type ${it.bookingType} does not match the requested type ${request.bookingType}." } }
+      .also { require(it.statusCode != StatusCode.CANCELLED) { "Video booking $videoBookingId is already cancelled, and so cannot be amended" } }
+      .also { require(it.appointments().all { a -> a.start().isAfter(now()) }) { "Video booking $videoBookingId has already started, and so cannot be amended" } }
+
+    val prisoner = request.prisoner().validate()
+
+    return booking.apply {
+      probationMeetingType = request.probationMeetingType!!.name
+      comments = request.comments
+      videoUrl = request.videoLinkUrl
+      this.amendedBy = amendedBy.username
+      amendedTime = now()
+    }
+      .also { thisBooking -> thisBooking.removeAllAppointments() }
+      .also { thisBooking -> appointmentsService.createAppointmentForProbation(thisBooking, request.prisoner(), amendedBy) }
+      .also { thisBooking -> videoBookingRepository.saveAndFlush(thisBooking) }
+      .also { thisBooking -> request.saveAdditionalDetailsFor(thisBooking) }
+      .also { thisBooking -> bookingHistoryService.createBookingHistory(HistoryType.AMEND, thisBooking) }
+      .also { thisBooking -> log.info("BOOKINGS: probation team booking ${thisBooking.videoBookingId} amended") } to prisoner
+  }
+
+  // We will only be creating appointments for one single prisoner as part of the initial rollout.
+  private fun AmendVideoBookingRequest.prisoner() = prisoners.first()
+
+  private fun PrisonerDetails.validate(): Prisoner {
+    // We are not checking if the prison is enabled here as we need to support prison users also.
+    // Our UI should not be sending disabled prisons though.
+    prisonRepository.findByCode(prisonCode!!) ?: throw EntityNotFoundException("Prison with code $prisonCode not found")
+    return prisonerValidator.validatePrisonerAtPrison(prisonerNumber!!, prisonCode).toPrisonerDetails()
+  }
+
+  private fun AmendVideoBookingRequest.saveAdditionalDetailsFor(booking: VideoBooking) {
+    additionalBookingDetails!!.let { details ->
```

Comment:
This has to change due to scope change.  Additional details need to be optional.  Will address in separate PR.